nomad/operator_endpoint_test.go

```diff
@@ -1188,3 +1189,183 @@ func TestOperator_SnapshotRestore_ACL(t *testing.T) {
+	err = state.UpsertNode(structs.MsgTypeTestSetup, 1001, outdatedNode)
+	must.NoError(t, err)
+
+	// Create non-default namespace.
+	ns := mock.Namespace()
+	state.UpsertNamespaces(1002, []*structs.Namespace{ns})
+
+	// Register Vault jobs, one with and another without workload identity.
+	jobNoWID := mock.Job()
+	jobNoWID.TaskGroups[0].Tasks[0].Vault = &structs.Vault{
+		Cluster:  "default",
+		Policies: []string{"test"},
+	}
+	// Add multiple tasks and groups to make sure we don't have duplicate jobs
+	// in the result.
+	jobNoWID.TaskGroups[0].Tasks = append(jobNoWID.TaskGroups[0].Tasks, jobNoWID.TaskGroups[0].Tasks[0].Copy())
+	jobNoWID.TaskGroups[0].Tasks[1].Name = "task-1"
+	jobNoWID.TaskGroups = append(jobNoWID.TaskGroups, jobNoWID.TaskGroups[0].Copy())
+	jobNoWID.TaskGroups[1].Name = "tg-1"
+
+	err = state.UpsertJob(structs.MsgTypeTestSetup, 1003, nil, jobNoWID)
+	must.NoError(t, err)
+
+	jobNoWIDNonDefaultNS := mock.Job()
+	jobNoWIDNonDefaultNS.Namespace = ns.Name
+	jobNoWIDNonDefaultNS.TaskGroups[0].Tasks[0].Vault = &structs.Vault{
+		Cluster:  "default",
+		Policies: []string{"test"},
+	}
+	err = state.UpsertJob(structs.MsgTypeTestSetup, 1004, nil, jobNoWIDNonDefaultNS)
+	must.NoError(t, err)
+
+	jobWithWID := mock.Job()
+	jobWithWID.TaskGroups[0].Tasks[0].Vault = &structs.Vault{
+		Cluster: "default",
+	}
+	jobWithWID.TaskGroups[0].Tasks[0].Identities = []*structs.WorkloadIdentity{{
+		Name: "vault_default",
+	}}
+	err = state.UpsertJob(structs.MsgTypeTestSetup, 1005, nil, jobWithWID)
+	must.NoError(t, err)
+
+	// Create allocs for the jobs.
+	allocJobNoWID := mock.Alloc()
+	allocJobNoWID.Job = jobNoWID
+	allocJobNoWID.JobID = jobNoWID.ID
+	allocJobNoWID.NodeID = node.ID
+
+	allocJobWithWID := mock.Alloc()
+	allocJobWithWID.Job = jobWithWID
+	allocJobWithWID.JobID = jobWithWID.ID
+	allocJobWithWID.NodeID = node.ID
+
+	err = state.UpsertAllocs(structs.MsgTypeTestSetup, 1006, []*structs.Allocation{allocJobNoWID, allocJobWithWID})
+	must.NoError(t, err)
+
+	// Create Vault token accessor for job without Vault identity and one that
+	// is no longer used.
+	tokenJobNoWID := mock.VaultAccessor()
+	tokenJobNoWID.AllocID = allocJobNoWID.ID
+	tokenJobNoWID.NodeID = node.ID
+
+	tokenUnused := mock.VaultAccessor()
+	err = state.UpsertVaultAccessor(1007, []*structs.VaultAccessor{tokenJobNoWID, tokenUnused})
+	must.NoError(t, err)
+
+	// Make request.
+	args := &structs.UpgradeCheckVaultWorkloadIdentityRequest{
+		QueryOptions: structs.QueryOptions{
+			Region:    "global",
+			AuthToken: node.SecretID,
+		},
+	}
+	var resp structs.UpgradeCheckVaultWorkloadIdentityResponse
+	err = msgpackrpc.CallWithCodec(codec, "Operator.UpgradeCheckVaultWorkloadIdentity", args, &resp)
+	must.NoError(t, err)
+	must.Eq(t, 1007, resp.Index)
+
+	// Verify only jobs without Vault identity are returned.
+	must.Len(t, 2, resp.JobsWithoutVaultIdentity)
+	must.SliceContains(t, resp.JobsWithoutVaultIdentity, jobNoWID.Stub(nil, nil), must.Cmp(cmpopts.IgnoreFields(
```

Comment:
I knew somebody would use `must.Cmp` one day!! :partying_face:

Reply:
I've used them a in a few places while working on node pools. Very handy for ignoring these Raft indexes that we usually don't care about 😄 

https://github.com/search?q=repo%3Ahashicorp%2Fnomad%20%22must.Cmp%22&type=code